frontend/src/pages/IntegrationAuthCallback/IntegrationAuthCallbackPage.tsx

```diff
@@ -286,14 +286,19 @@ const WorkspaceIntegrationCallback = ({
 	type: string
 	addIntegration: (code: string) => Promise<unknown>
 }) => {
+	const codeSessionStorageKey = 'integration-callback-code'
```

Comment:
This component and its parent are re-rendered multiple times. Don't know if that was intended.
The net effect however is that the IIFE executed in the useEffect hook for the WorkspaceIntegrationCallback
component is executed twice - calling ```await addIntegration(code) twice ```

For my workspace integration, this meant that I got two notifications - once succeeding while the other fails.
That is because my authorization_code cannot be used multiple times. The failed notification is a false negative in
this case and

So I added a way for us to store the usedAuthorization code in sessionStorage so that we prevent calling addIntegration on the same code when the component renders. Feels a bit hacky though.

I looked into other ways to ensure that the IIFE executed just once but couldn't come up with a better solution
thus far. Open to other alternatives

Reply:
Could the rendering multiple times be a side effect of how we use `React.StrictMode` in our dev environments? Anyway I think this is an ok workaround